mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -1909,7 +1909,7 @@ llvm::omp::OMPTgtExecModeFlags TargetOp::getKernelExecFlags() {
 
   // Detect Generic-SPMD: target-teams-distribute[-simd].
   if (numWrappers == 1) {
-    if (!isa<DistributeOp>(innermostWrapper))
+    if (!isa<DistributeOp>(innermostWrapper) && !isa<LoopOp>(innermostWrapper))
```

Comment:
This is slightly incorrect, because it will cause `target teams loop` to be tagged as Generic-SPMD instead of SPMD. Perhaps you can set a local `llvm::omp::OMPTgtExecModeFlags` variable to `OMP_TGT_EXEC_MODE_GENERIC_SPMD` or `OMP_TGT_EXEC_MODE_SPMD` based on whether the wrapper is `omp.distribute` or `omp.loop` and return it if all conditions below are met, instead of it always being `OMP_TGT_EXEC_MODE_GENERIC_SPMD`.

Also update the comment above, to prevent it from becoming misleading.